onnx_xla/onnx_xla_interface.cc

```diff
@@ -267,3 +277,79 @@ ONNXIFI_SYMBOL_NAME(onnxReleaseGraph)(onnxGraph graph) {
   }
   ONNXIFI_CATCH_EXCPETION();
 }
+
+//Initialize event by creating EventControl object on the heap
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxInitEvent)(onnxBackend backend, onnxEvent* event) {
+  try {
```

Comment:
`event` must be set to `NULL`/`nullptr` if the function fails. I suggest to do it on the first line of the function, and then overwrite it with actual event pointer, if the function succeeded.

Reply:
@varunjain99 But if `backend` is null then the current logic will directly return without setting event to null

Reply:
right - order is changed in my commit